spec/services/moderator/manage_activity_and_roles_spec.rb

```diff
@@ -4,95 +4,199 @@
+        expect(user.roles.pluck(:name)).to include("tag_moderator") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).not_to include("tag_moderator") # confirm assumptions
+      end
+    end
+
+    context "when user is in admin role" do
+      before { user.add_role(:admin) }
+
+      it "adding #{status} ignores the admin role" do
+        expect(user.roles.pluck(:name)).to include("admin") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).to include("admin") # confirm assumptions
+      end
+    end
+
+    context "when user is in super_moderator role" do
+      before { user.add_role(:super_moderator) }
+
+      it "adding #{status} ignores the super_moderator role" do
+        expect(user.roles.pluck(:name)).to include("super_moderator") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).to include("super_moderator") # confirm assumptions
+      end
+    end
+
+    context "when user is in tech_admin role" do
+      before { user.add_role(:tech_admin) }
+
+      it "adding #{status} ignores the tech_admin role" do
+        expect(user.roles.pluck(:name)).to include("tech_admin") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).to include("tech_admin") # confirm assumptions
+      end
+    end
+
+    context "when user is in limited role" do
+      before { user.add_role(:limited) }
+
+      it "adding #{status} ignores the limited role" do
+        expect(user.roles.pluck(:name)).to include("limited") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).to include("limited") # confirm assumptions
+      end
+    end
+
+    context "when user is in warned role" do
+      before { user.add_role(:warned) }
+
+      it "adding #{status} ignores the warned role" do
+        expect(user.roles.pluck(:name)).to include("warned") # confirm assumptions
+        manage_roles_for user, user_status: status
+        expect(user.roles.pluck(:name)).to include("warned") # confirm assumptions
+      end
+    end
   end
 
-  it "assigns trusted role to user that's updated to super admin" do
-    described_class.handle_user_roles(
-      admin: admin,
-      user: user,
-      user_params: { note_for_current_role: "Upgrading to super admin", user_status: "Super Admin" },
-    )
-    expect(user.super_admin?).to be true
-    expect(user.has_trusted_role?).to be true
+  it_behaves_like "elevated role", "Admin"
+  it_behaves_like "elevated role", "Super Moderator"
+  it_behaves_like "elevated role", "Resource Admin: Tag"
+  it_behaves_like "elevated role", "Super Admin"
+  it_behaves_like "elevated role", "Trusted"
+  it_behaves_like "elevated role", "Good standing"
+  it_behaves_like "elevated role", "Tech Admin"
```

Comment:
I know it's only in these specs, but I am a bit iffy about calling _all_ of these roles "elevated roles" - elsewhere we use "elevated user" to refer to only admins and super moderators?

Reply:
For what it's worth, I got the "elevated" language from [here](https://github.com/forem/forem/blob/915b3eaca4e70e77943facd3fd8b27f4ae241fe5/app/services/moderator/manage_activity_and_roles.rb#L61-L109) and it _mostly_ matches the roles we're using there. There _are_ inconsistencies, but, from what I can tell, they're _mostly_ that we aren't using the utility method consistently (mostly for historical/legacy reasons). 

For example, "Resource Admin" doesn't invoke `assigned_elevated_role`, but duplicates everything it does except clearing a cache. (And, even then, because we check `Rack::Attack::ADMIN_ROLES`, it's not even as if _all_ the callers assume that cache is being cleared. I _don't_ see a reason "Resource Admin" _can't_ call `assign_elevated_role`, except that it does some _extra_ stuff `constantize`ing the role name, so we duplicated the parts that ought to be common behavior, rather than special-casing the actually special cases?)

I am approximately ~95% confident that these are all "elevated roles" in our system. I would _really_ like to find time to refactor `Moderator::ManageActivityAndRoles` to be more consistently systematic in its approach, but it seemed like too much for this PR.